rust/hyperlane-core/src/rpc_clients/fallback.rs

```diff
@@ -0,0 +1,189 @@
+use async_rwlock::RwLock;
+use async_trait::async_trait;
+use derive_new::new;
+use std::{
+    fmt::Debug,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+use tracing::info;
+
+use crate::ChainCommunicationError;
+
+/// Read the current block number from a chain.
+#[async_trait]
+pub trait BlockNumberGetter: Send + Sync + Debug {
+    /// Latest block number getter
+    async fn get(&self) -> Result<u64, ChainCommunicationError>;
```

Comment:
nit: don't feel strongly as maybe this is a common Rust pattern to rely more on the trait name than the function name to describe a function, but wondering if this should be `get_block_number` or something w/ more clarity in the function name

Reply:
totally had meant to change this, thanks for pointing out